### DISCUSSION TITLE
Dynamically creating and registering properties in ReadOnlyBase<T>?

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#3600] [Questions]
Has anyone experimented with read-only objects with dynamic properties?

I have a situation where I want an info object which I can bind to a `DataGridView` in WinForms .NET 4.8, but the properties will all be user-defined. This project is using CSLA 4.11.2.

Alternatively, I could define something like 200 properties at compile time which pass through to a dictionary or list.

**Comment by michaelcsikos** [comment]
If we have to inherit from `DynamicObject` I can't see how this would be possible.

**Reply by rockfordlhotka** [reply]
I suspect it would be a fair amount of work. Still, the only real requirement for base types in CSLA now is that they implement `IMobileObject`.

To do what you describe your type will have to maintain a dictionary of property names and values I assume? Those map to the dynamic properties exposed by the type?

You can probably implement `IMobileObject` to serialize/deserialize those property values - especially if they are all "primitive" types directly supported by MobileFormatter, it shouldn't be too hard.

**Comment by ajohnstone-ks** [comment]
I've had to do something similar.  Basically we use Csla and the object has a MobileDictionary<string, string>, then on the client we use the dictionary to build a DataTable and bind the grid to the data table.  If having the values as a string is a problem you could return a string with xml, and then parse the xml.  That way you can include the data type of each value as well.  But I don't think you'd be able to have BOs with editable properties in the grid.
